### ISSUE TITLE
Compatibility with prettier 3

### ISSUE THREAD

**Issue by sawa-ko** [victorgarciaesgi/nuxt-typed-router#106]
**Describe the bug**
When the package is used together with prettier v3 an internal error occurs in the package.

```
// my package.json
"prettier": "^3.0.0"
```

**Expected behavior**
No problem with prettier.

**Screenshots**
```ts
> dashboard@1.9.1 postinstall /workspaces/dashboard
> nuxt prepare

Nuxi 3.6.2                                                                                                                                         3:54:56 PM
ℹ Using html-validate to validate server-rendered HTML                                                                                            3:54:57 PM
ℹ Using default Tailwind CSS file                                                                                                nuxt:tailwindcss 3:54:58 PM

 WARN  Vue I18n configuration file does not exist at i18n.config. Skipping...                                                         @nuxtjs/i18n 3:54:58 PM

ℹ nuxt-auth setup starting                                                                                                              nuxt-auth 3:54:59 PM
✔ nuxt-auth setup done                                                                                                                  nuxt-auth 3:54:59 PM

 ERROR  ✖ Error while formatting the output                                                                                                       3:55:00 PM
TypeError: Cannot read properties of undefined (reading 'startsWith')


 ERROR  Cannot read properties of undefined (reading 'startsWith')                                                                                 3:55:00 PM

  at printAngular (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:24775:22)
  at printPathNoParens (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:29687:25)
  at Object.genericPrint [as print] (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:29629:23)
  at callPluginPrintFunction (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:8601:26)
  at mainPrintInternal (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:8550:22)
  at mainPrint (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:8537:18)
  at printAstToDoc (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:8529:18)
  at coreFormat (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:8837:20)
  at formatWithCursor2 (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:9021:18)
  at node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:38183:12
  at format (node_modules/.pnpm/prettier@2.8.8/node_modules/prettier/index.js:38197:12)
  at formatOutputWithPrettier (node_modules/.pnpm/nuxt-typed-router@3.2.4/node_modules/nuxt-typed-router/dist/module.mjs:1252:30)
  at async processPathAndWriteFile (node_modules/.pnpm/nuxt-typed-router@3.2.4/node_modules/nuxt-typed-router/dist/module.mjs:1274:29)
  at async Promise.all (index 0)
  at async saveGeneratedFiles (node_modules/.pnpm/nuxt-typed-router@3.2.4/node_modules/nuxt-typed-router/dist/module.mjs:1363:3)
  at async node_modules/.pnpm/nuxt-typed-router@3.2.4/node_modules/nuxt-typed-router/dist/module.mjs:1580:7
  at async Object.getContents (node_modules/.pnpm/nuxt@3.6.2_@types+node@20.4.1_eslint@8.44.0/node_modules/nuxt/dist/index.mjs:698:9)
  at async node_modules/.pnpm/nuxt@3.6.2_@types+node@20.4.1_eslint@8.44.0/node_modules/nuxt/dist/index.mjs:3652:22
  at async Promise.all (index 31)
  at async generateApp (node_modules/.pnpm/nuxt@3.6.2_@types+node@20.4.1_eslint@8.44.0/node_modules/nuxt/dist/index.mjs:3651:3)
  at async _applyPromised (node_modules/.pnpm/perfect-debounce@1.0.0/node_modules/perfect-debounce/dist/index.mjs:54:10)

 ELIFECYCLE  Command failed with exit code 1.
```

**Environnement infos**

Run `nuxi infos`

- Operating System: Linux
- Node Version:     v20.3.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  pnpm@8.6.2
- Builder:          vite
- User Config:      modules, imports, experimental, css, htmlValidator, app, runtimeConfig, colorMode, security, i18n, content, nuxtTypedRouter, ogImage, pinia, robots, auth, plausible
- Runtime Modules:  @nuxt/content@2.7.0, @nuxtjs/html-validator@1.5.1, @nuxtjs/fontaine@0.4.0, @nuxtjs/tailwindcss@6.8.0, @nuxtjs/color-mode@3.3.0, @nuxtjs/i18n@8.0.0-beta.13, @nuxt/image-edge@1.0.0-rc.1-28143901.afe4120, @nuxt/devtools@0.6.7, nuxt-security@0.13.1, nuxt-icon@0.4.2, nuxt-typed-router@3.2.4, @sidebase/nuxt-auth@0.6.0-beta.3, nuxt-og-image@2.0.9, @pinia/nuxt@0.4.11, @nuxtjs/robots@3.0.0, v-plausible@1.0.2
- Build Modules:    -

**Your `pages` folder structure**

Run `npx tree-node-cli {your page folder path}`
```
pages
├── index.vue
├── privacy.vue
├── profile
│   └── [id].vue
└── terms.vu
```

**Your nuxt.config.ts**

```ts
import { resolve } from 'node:path';

import { defineNuxtConfig } from 'nuxt/config';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  modules: [
    '@nuxtjs/html-validator',
    '@nuxtjs/fontaine',
    '@nuxtjs/tailwindcss',
    '@nuxtjs/i18n',
    '@nuxt/devtools',
    'nuxt-typed-router',
  ],
  imports: {
    dirs: ['stores'],
  },
  experimental: {
    componentIslands: true,
    inlineSSRStyles: (id) => !id?.includes('entry'),
  },
  css: ['~/assets/main.css'],
  htmlValidator: {
    usePrettier: true,
    options: {
      rules: {
        'heading-level': ['warn', { allowMultipleH1: true }],
      },
      elements: [
        'html5',
        {
          a: {
            permittedDescendants: ['@flow', 'button'],
          },
          button: {
            permittedContent: ['@flow', 'div'],
          },
        },
      ],
    },
  },
  i18n: {
    baseUrl: siteUrl,
    defaultLocale: 'en',
    strategy: 'no_prefix',
    langDir: 'locales',
    locales: [
      {
        name: 'English',
        code: 'en',
        iso: 'en-US',
        file: 'en.json',
      },
      {
        name: 'Español',
        code: 'es',
        file: 'es.json',
        iso: 'es-ES',
      },
    ],
    detectBrowserLanguage: {
      alwaysRedirect: true,
      redirectOn: 'all',
      cookieKey: 'neko-lang',
    },
  },
  nuxtTypedRouter: {
    strict: true,
  },
});
```

**Comment by sawa-ko** [comment]
Temporarily solved with (pnpm):

```json
{
	"pnpm": {
		"overrides": {
			"prettier": "3.0.0"
		}
	}
}

```

**Comment by victorgarciaesgi** [comment]
Hi! Yeah I think I will ditch prettier, it will cause too much compatibility issues and it's not worth it to have generated code prettified

**Comment by victorgarciaesgi** [comment]
Fixed in `3.2.5`
